### ISSUE TITLE
finish npm as build-tool

### ISSUE THREAD

**Issue by jzacsh** [jzacsh/doodle#1]
Previously this codebase was working as a [manually managed HTML,js tree](https://github.com/jzacsh/doodle/blob/10884d171e2b749c135609b71bd7445e4de9766e/index.html#L11-L28) at commit 10884d171e2b749c135609b71bd7445e4de9766e

However, as of e3ae97cca03fed13f20c2aee390e46b1178bb25a the process to move to commonjs modules was started, but left in a broken state. Currently, somehow, running the following:

``` bash
$ npm run reserve
```

Somehow results in angular executing **after** [doodle_app.js](https://github.com/jzacsh/doodle/blob/master/src/doodle_app.js) - _the entry point to this web app_ - executes.
In other words, [this line](https://github.com/jzacsh/doodle/blob/e3ae97cca03fed13f20c2aee390e46b1178bb25a/src/doodle_app.js#L10) results in:

``` javascript
Uncaught TypeError: Cannot read property 'module' of undefined  app.js:11
  (anonymous function)  app.js:30
  1../doodle_page       app.js:1
  s                     app.js:1
  e                     app.js:1
  (anonymous function)  app.js:1
```

**Comment by jzacsh** [comment]
fixed by commit 1c676f21f8f233d2c214de5885784208b55d9e63
